added usage example to pad_to_bounding_box()

generationzcode opened the pull request:
I added a usage example to tf.image.pad_to_bounding_box() function. I tested it on colab so I know there aren't syntax errors.😄🖖

generationzcode committed:
added usage example to pad_to_bounding_box()

generationXcode commented:
I signed it!

kyscg commented:
@googlebot I signed it!

kyscg commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
       negative.
+      
+  Usage example:
+    import tensorflow as tf
```
```suggestion
    
```

kyscg commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
```
```suggestion
  Usage example:
  ```python
```

kyscg commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+      
+  Usage example:
+    import tensorflow as tf
+    x=tf.random.uniform((28,28,1))
```
```suggestion
    >>> x = tf.random.uniform((28, 28, 1))
```

kyscg commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+  Usage example:
+    import tensorflow as tf
+    x=tf.random.uniform((28,28,1))
+    tf.image.pad_to_bounding_box(x,28,28,100,100)
```
```suggestion
    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
```

kyscg commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+    import tensorflow as tf
+    x=tf.random.uniform((28,28,1))
+    tf.image.pad_to_bounding_box(x,28,28,100,100)
+    
```
```suggestion
    ```
    
```

kyscg reviewed (changes requested):
There is no need to import tensorflow everywhere, please make these changes

generationzcode committed:
Update tensorflow/python/ops/image_ops_impl.py

Co-Authored-By: Kilaru Yasaswi Sri Chandra Gandhi <yasaswisrichandragandhi@gmail.com>

generationzcode committed:
Update tensorflow/python/ops/image_ops_impl.py

Co-Authored-By: Kilaru Yasaswi Sri Chandra Gandhi <yasaswisrichandragandhi@gmail.com>

generationzcode committed:
Update tensorflow/python/ops/image_ops_impl.py

Co-Authored-By: Kilaru Yasaswi Sri Chandra Gandhi <yasaswisrichandragandhi@gmail.com>

generationzcode committed:
Update tensorflow/python/ops/image_ops_impl.py

Co-Authored-By: Kilaru Yasaswi Sri Chandra Gandhi <yasaswisrichandragandhi@gmail.com>

generationzcode committed:
Update tensorflow/python/ops/image_ops_impl.py

Co-Authored-By: Kilaru Yasaswi Sri Chandra Gandhi <yasaswisrichandragandhi@gmail.com>

generationXcode commented:
Done it!

alextp commented on tensorflow/python/ops/image_ops_impl.py:
````diff
@@ -792,6 +792,14 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
+    ```
````
Assert something about the result to make this a doctest? Like print its shape at least?

generationXcode replied:
Ok!

generationXcode replied:
done!😊

alextp replied:
Then put the resulting shape in the comment so this runs as a doctest

generationXcode replied:
done! If there are any other changes needed please say so. I can't wait to see this merged😊

generationzcode committed:
edited usage example of pad to bounding box

generationzcode committed:
added shape comment to doctest

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -792,6 +792,15 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    >>> # TensorShape([100, 100, 1])
```
Why is the result commented? You can remove the `>>> #` and this will make the doctest work.

generationXcode replied:
Did it!

generationzcode committed:
Uncommented TensorShape([100, 100, 1])

generationzcode committed:
Uncommented TensorShape([100, 100, 1])

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -773,7 +773,7 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
   This op does nothing if `offset_*` is zero and the image already has size
   `target_height` by `target_width`.
-
+  
```
Please remove all trailing whitespace

mihaimaruseac replied:
The whitespace is not removed
![Screenshot from 2020-01-04 13-14-23](https://user-images.githubusercontent.com/323199/71771688-3982e900-2ef4-11ea-80f4-7f94a77e333a.png)

generationXcode replied:
oh sorry.

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
````diff
@@ -792,6 +792,15 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
       negative.
+      
+  Usage example:
+  ```python
````
No need to add the ` ```python` and ` ``` ` markers. Just have an empty line after `Usage example:` and after the code block. Move the doctest above, just before the `Args:` section

generationzcode committed:
Update image_ops_impl.py

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -790,8 +790,17 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+      
+  Usage example:
+  
+  python
```
Why is this `python` here?

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -790,8 +790,17 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    >>> TensorShape([100, 100, 1])
```
Why are these here?

generationXcode replied:
Its the usage example

generationXcode replied:
But I think you might be referring to the `TensorShape([100, 100, 1])`... Umm in this commit I removed the `>>>` that comes before it, so everyone knows its the output.

mihaimaruseac reviewed (changes requested):
Please consult https://www.tensorflow.org/community/contribute/docs_ref and follow the guideliness from there

generationzcode committed:
Update image_ops_impl.py

Removed **all** trailing whitespace, removed `python from the usage example, moved the usage example above the args and more`

generationXcode commented:
Done!

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
 def pad_to_bounding_box(image, offset_height, offset_width, target_height,
                         target_width):
   """Pad `image` with zeros to the specified `height` and `width`.
-
```
Why did you remove this line?

generationXcode replied:
I fixed it.😊

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
   Adds `offset_height` rows of zeros on top, `offset_width` columns of
   zeros on the left, and then pads the image on the bottom and right
   with zeros until it has dimensions `target_height`, `target_width`.
-
```
Why did you remove this line?

generationXcode replied:
I fixed it.😊

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
   This op does nothing if `offset_*` is zero and the image already has size
   `target_height` by `target_width`.
-
+  Usage example:
```
Empy line above.

generationXcode replied:
I fixed it.😊

generationXcode replied:
I fixed it.😊

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    TensorShape([100, 100, 1])
```
Empty line below

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
-      `target_*` arguments, or either `offset_height` or `offset_width` is
-      negative.
+    `target_*` arguments, or either `offset_height` or `offset_width` is
+    negative.    
```
Don't add trailing whitespace.

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    TensorShape([100, 100, 1])
```
I would rather have `x` be a small constant with some nice values and then pad to a slightly larger box (instead of going from 28 to 100) and displaying the values instead of just the shape

generationXcode replied:
you mean `tf.constant`?

mihaimaruseac replied:
Yes

generationXcode replied:
Hi! I have no problem with the changing of the data, I just dont know how to display the values of a tensor. nvm I'll find out😊

mihaimaruseac replied:
Just run the test on your local machine. https://www.tensorflow.org/community/contribute/docs_ref#test_on_your_local_machine

mihaimaruseac replied:
This is still not done

generationzcode committed:
Update image_ops_impl.py

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -766,32 +766,36 @@ def _get_dim(tensor, idx):
                         target_width):
   """Pad `image` with zeros to the specified `height` and `width`.
-
+  
```
![dTr3qVQtZEr](https://user-images.githubusercontent.com/323199/71906884-d70e3080-311f-11ea-82e6-49d313010ee6.png)

mihaimaruseac reviewed (commented):
Don't forget to remove trailing whitespace

generationzcode committed:
Update image_ops_impl.py

generationzcode committed:
Update image_ops_impl.py

mihaimaruseac reviewed (changes requested):
Please change doctest from testing only the shape of the tensor to also testing the values, as told in comments

generationXcode commented:
@mihaimaruseac Do we use `tensor.eval()` to show the values? I tried it once, it gave me errors...

mihaimaruseac commented:
No need to `tensor.eval()` (though it would have been useful to also see what errors you were getting).

You can just do something similar to

```python
>>> x = tf.constant([2,3,4])
>>> y = tf.constant([4,5,6])
>>> tf.add(x, y)
<Tensor... [6,8,10]...>
```

(where I ellided a lot of details in the output, but you get the general idea)

generationXcode commented:
got it!

generationzcode committed:
Update image_ops_impl.py

mihaimaruseac commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -774,9 +774,19 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
+    ...        [[1,2,3],[4,5,6],[7,8,9]],
+    ...        [[1,2,3],[4,5,6],[7,8,9]]
+    ...        ])
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
```
Change the bounding box to smaller values so that we could print all the values in `numpy=array([[[...`?

generationXcode replied:
got it. I'll reduce the size of both the arrays.

generationzcode committed:
Update image_ops_impl.py

gbaned commented:
@generationXcode thank you, it is still failing doctest can you please check here for [logs](https://source.cloud.google.com/results/invocations/383d702a-33c9-42f2-94a3-0038329704f1/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests).

Please run the doctest locally as mentioned here in the [contributor guidelines](https://www.tensorflow.org/community/contribute/docs_ref).

Also, can you please help to fix Ubuntu Sanity errors? Thanks!

generationXcode commented:
Hi, sorry for such a late reply (its my computer time now). Umm I don't know much about these checks but I am doing what you said and running the doctest locally, presently I'm downloading the tensorflow repo

generationzcode committed:
Update image_ops_impl.py

gbaned commented:
@generationXcode  it is still failing doctest can you please check here for [logs](https://source.cloud.google.com/results/invocations/4de04c11-2db2-48f6-8ced-b4847ba81db5/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests). Thanks!

generationXcode commented:
@gbaned is there any way we can run the doctest again? Thanks!

mihaimaruseac commented:
Unfortunately we can only trigger all tests at once. Retriggered them now

generationXcode commented:
Thanks!

generationXcode commented:
it seems like the Ubuntu CPU test is contradicting the other results https://source.cloud.google.com/results/invocations/a46ee8e1-2790-41ad-8f78-216803471602/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests
If I solve this problem then the other tests will fail...

generationzcode commented:
Hi, I know this pull request isn't closed, but GCI is and I'd like to thank all of the mentors for helping me during the contest and also the reviewers of the pull requests I made.😊
